lib/src/main/kotlin/com/swmansion/starknet/provider/rpc/JsonRpcProvider.kt

```diff
@@ -44,65 +58,81 @@ class JsonRpcProvider(
         }
     }
 
-    private fun buildRequestJson(method: String, paramsJson: JsonElement): Map<String, JsonElement> {
-        val map = mapOf(
-            "jsonrpc" to JsonPrimitive("2.0"),
-            "method" to JsonPrimitive(method),
-            "id" to JsonPrimitive(0), // It is not used anywhere
-            "params" to paramsJson,
-        )
-
-        return JsonObject(map)
-    }
-
     private fun <T> buildRequest(
         method: JsonRpcMethod,
         paramsJson: JsonElement,
         responseSerializer: KSerializer<T>,
     ): HttpRequest<T> {
-        val requestJson = buildRequestJson(method.methodName, paramsJson)
+        val jsonRpcRequest = JsonRpcRequest(
+            id = 0,
+            jsonRpc = "2.0",
+            method = method.methodName,
+            params = paramsJson,
+        )
 
-        val payload =
-            HttpService.Payload(url, "POST", emptyList(), requestJson.toString())
+        return HttpRequest(url, jsonRpcRequest, responseSerializer, deserializationJson, httpService)
+    }
+
+    fun <T> batchRequests(requests: List<HttpRequest<T>>): BatchHttpRequest<T> {
+        require(requests.isNotEmpty()) { "Please provide requests while creating a batching requests" }
+
+        return buildBatchRequest(requests)
```

Comment:
Seems like it can be inlined if you refactor vararg version of the method?

Reply:
Could you explain more about what you mean by "refactoring the vararg version of the method"? 😄

Reply:
> Could you explain a bit more what you mean 😄 ? 

`buildBatchRequest ` can be removed with its contents inlined here, as it's not used anywhere else

Reply:
Got it. I've applied proper refactor. 🫡